File: 14889.cpp
#include <cstdio>
#include <array>
#include <vector>
using namespace std;

int N;
array<array<int, 20>, 20> stats;
vector<int> combinations;

void pick(int count, int picked) {
  if (count==(int)(N/2)) {
    combinations.push_back(picked);
    return;
  }
  for (int i = picked==0 ? 0; i<N; ++i) {
    picked.push_back(i);
    pick(count+1);
    picked.pop_back();
  }
}

int main(void) {
  scanf("%d", &N);
  for (int i=0; i<N; ++i) {
    for (int j=0; j<N; ++j) {
      scanf("%d", &stats[i][j]);
    }
  }
  pick(0);
  // int minimum = 100*2*20;
  // for (int i=0; i<combinations.size(); ++i) {
  // }
}

File: 14889.py
N = int(input())
stat = []
for _ in range(N):
  stat.append(list(map(int, input().split())))

def statOf(team)->int:
  result = 0
  for i in team:
    for j in team:
      result+=stat[i][j]
  return result
everyone = set(range(0, N))
combinations = []
picked = []

def pick(count):
  if count==N//2:
    combinations.append(set(picked))
    return
  for i in range(0 if not picked else picked[-1]+1, N):
    picked.append(i)
    pick(count+1)
    picked.remove(i)
pick(0)
minimum = 100*2*20
for c in combinations:
  minimum = min(minimum, abs(statOf(c)-statOf(everyone-c)))
print(minimum)
